RecoPixelVertexing/PixelTrackFitting/plugins/PixelTrackProducerFromSoA.cc

```diff
@@ -133,8 +133,8 @@ void PixelTrackProducerFromSoA::produce(edm::StreamID streamID,
   const auto &tsoa = *iEvent.get(tokenTrack_);
 
   auto const *quality = tsoa.qualityData();
-  auto const &fit = tsoa.stateAtBS;
-  auto const &hitIndices = tsoa.hitIndices;
+  auto const &fit = tsoa.stateAtBS();
```

Comment:
as noted before, (and discussed already also in CORE, there is no agreement that this is a better syntax for SOA.

Reply:
This was also a consequence of the 4.25 rule which forces to choose between direct member access in struct style or using accessors, and this class was a in-between.

Reply:
There are very good reason why some members are directly accessed and other through a get functon. For instance it let user to think was is going on! It avoids obfuscations.

Reply:
I think the question meant: is there any chance we want this to ever be a `reinterpret_cast` ?
If not using a `static_cast` is more accurate than a C cast (that can apply all of `static_cast`, `reinterpret_cast` and `const_cast` at once).

Reply:
in that specific case we want the compiler to emit a conversion (by value!). We want a temporary float created there!
equivalent to  ```float tmp = f; ``` no more, no less. nothing fancy. As in good old C code (or assembler).
I prefer to introduce such an intermediate statement than use a C++ cast that has not place in math expressions.

Reply:
The only reinterpret_cast accepted is  reinterpret_cast<long long int&>(x) and this is funny as it is equivalent to type punning that is undef behavioir. in C++20 they have introduced bit_cast that is equivalent to a memcpy which is the correct semantic. In short  nobody uses ```reinterpret_cast``` among intrinsic type.